infrastructure/smart-contracts/contracts/staking-v2/StakingPool.sol

```diff
@@ -209,26 +221,41 @@ contract StakingPool is AccessControlUpgradeable {
 
     function stakeKeys(
         address owner,
-        uint256[] memory keyIds
+        uint256 keyAmount
     ) external onlyRole(DEFAULT_ADMIN_ROLE) {
-        uint256 keyLength = keyIds.length;
-        for (uint i = 0; i < keyLength; i++) {
-			// Update indexes of this user's staked keys
-            keyIdIndex[keyIds[i]] = stakedKeysOfOwner[owner].length;
-            stakedKeysOfOwner[owner].push(keyIds[i]);
-
-			// Update indexes of the pool's staked keys
-			stakedKeysIndices[keyIds[i]] = stakedKeys.length;
-			stakedKeys.push(keyIds[i]);
+
+        // Handle migration for key amounts
+        if(!hasMigratedToKeyAmount[owner]){
+            stakedKeyAmounts[owner] = stakedKeysOfOwner[owner].length;
+            hasMigratedToKeyAmount[owner] = true;
         }
```

Comment:
Is the intention to do a lazy migration of this data? This snippet is present in several of the other functions.

Reply:
It is, I added a function to remove all the duplicates.